CDLUC3/dmsp_frontend_prototype
# Added two new TypeAhead versions to the styleguide

## Description

We will need TypeAhead/Autosuggest components for pages like the "Register" page where users have to search for an Institution.

- I added two TypeAhead components to the app: the first one does not include an "Other" option and the second one has it, and expects a "setOtherField" method to be passed to it that can be set to true when a user selects the "Other" field. That way the calling component can know to show the additional "Other" field.

- I remove the files related to the pilot project dmp landing page because it was breaking the build, and was added primarily for testing purposes.
- I added the use of graphql-codegen to generate the types for the graphql queries.
- Updated README.md file

Fixes # [10](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/10)

## Type of change
Please delete options that are not relevant

## How Has This Been Tested?
The new TypeAheads were tested against the backend server GraphQL API for `affiliations` after they were added to the styleguide.

- Start the app along with the dmsp_backend_prototype `development` branch.
- Navigate to http://localhost:3000/styleguide and scroll down to the `Forms fields` section.
- Some tests to run
-- Enter "alaska" into input field. Should show results in dropdown
-- Select an option from the dropdown either using "Click" or "Enter" key. The input value to change to the selected one
-- Navigate through the dropdown using the up and down arrow keys
-- Enter a nonsensical value like "asdfadf". Try and tab out of the field. You should be restricted from doing so.



Also, unit tests for the components were added.


## Checklist:
